server/conn.go

```diff
@@ -153,6 +153,10 @@ func (cc *clientConn) Close() error {
 	delete(cc.server.clients, cc.connectionID)
 	connections := len(cc.server.clients)
 	cc.server.rwlock.Unlock()
+	return closeConn(cc, connections)
```

Comment:
Will `Close` be called twice?

Reply:
Normally won't. 
https://github.com/pingcap/tidb/blob/master/server/conn.go#L458 There will check connection status to avoid double close.